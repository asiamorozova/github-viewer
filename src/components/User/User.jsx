import React from 'react';
import PropTypes from 'prop-types';

const User = ({ login, followers_url, following_url, url, avatar_url }) => (
  <figure>
    <img src={avatar_url} alt={login}/>
    <figcaption>
      <p>{login}</p>
      <p>Followes:{followers_url}</p>
      <p>Following:{following_url}</p>
      <p>{url}</p>
    </figcaption>
  </figure>
);

User.propTypes = {
  login: PropTypes.string.isRequired,
  followers_url: PropTypes.string.isRequired,
  following_url: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  avatar_url: PropTypes.string.isRequired

};

export default User;

