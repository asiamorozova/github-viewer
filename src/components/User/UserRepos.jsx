import React from 'react';
import PropTypes from 'prop-types';
import UserRepo from './UserRepo';

const UserRepos = ({ repos }) => {
  const repoData = repos.map(repo => <UserRepo key={repo.id} {...repo} />);
  return (
    <ul>
      {repoData}
    </ul>
  );
};

UserRepos.propTypes = {
  repos: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    html_url: PropTypes.string.isRequired
  }))
};

export default UserRepos;
