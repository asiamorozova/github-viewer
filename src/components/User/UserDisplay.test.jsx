import React from 'react';
import { shallow } from 'enzyme';
import User from './UserDisplay';

describe('User', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<User login='asiamorozova'
    
      followers_url='1' 
      following_url='1' 
      url='https://github.com/asiamorozova' 
      avatar_url='https://avatars0.githubusercontent.com/u/57855398?v=4' />);
    expect(wrapper).toMatchSnapshot();
  });
});
