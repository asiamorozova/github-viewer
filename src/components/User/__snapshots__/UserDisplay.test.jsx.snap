// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User matches a snapshot 1`] = `
<figure>
  <img
    alt="asiamorozova"
    src="https://avatars0.githubusercontent.com/u/57855398?v=4"
  />
  <figcaption>
    <p>
      asiamorozova
    </p>
    <p>
      Followers:
      1
    </p>
    <p>
      Following:
      1
    </p>
    <p>
      https://github.com/asiamorozova
    </p>
  </figcaption>
</figure>
`;
